### PR TITLE
Convert dash to underscore when generating apps

### LINKED ISSUE #391: Lotus generate app is not fixing invalid app-names
For app-names containing dashes, `lotus generate app` results in an exception (see the following steps to reproduce: https://gist.github.com/florianb/f8809b55267d4fcea6c7).

I'm currently trying to figure out where the overall-application-name is handled different to the names of additional apps. I guess it is intended to use the same naming-strategy as with `lotus new`.

I'll prepare a PR if wanted.

### PR DESCRIPTION
**Pull request by AlfonsoUceda**
This PR fix #391

### CODE REVIEW

File: test/application_name_test.rb [hunk 1]
```diff
@@ -8,12 +8,12 @@
     describe "#to_s" do
       it "renders downcased" do
         application_name = Lotus::ApplicationName.new("MY-APP")
-        application_name.to_s.must_equal "my-app"
+        application_name.to_s.must_equal "my_app"
       end
 
       it "renders trimmed" do
         application_name = Lotus::ApplicationName.new(" my-app ")
```
Comment (joneslee85) [inline test/application_name_test.rb]: remove empty spaces after (

Comment (joneslee85) [inline test/application_name_test.rb]: and before )

File: test/commands/generate/app_test.rb [hunk 2]
```diff
@@ -34,6 +34,17 @@
       end
     end
 
+    it 'generate the application with valid ruby syntax for dasherized name' do
```
Comment (joneslee85) [inline test/commands/generate/app_test.rb]: s/generate/generates

**Comment by joneslee85** [comment]
+1

File: test/application_name_test.rb (hunk 1 above)
Comment (AlfonsoUceda) [inline test/application_name_test.rb]: Those spaces are intentioned for the test

Comment (joneslee85) [inline test/application_name_test.rb]: oh it's good then

**Comment by jodosha** [comment]
@AlfonsoUceda This looks good :+1: Please wait until CI is green (I've restarted single JRuby failures).

**Pull request merged**